patterns/karpenter-mng/main.tf

```diff
@@ -52,7 +52,13 @@ data "aws_ecrpublic_authorization_token" "token" {
   provider = aws.virginia
 }
 
-data "aws_availability_zones" "available" {}
+data "aws_availability_zones" "available" {
```

Comment:
Why are we changing this data source?

Reply:
If you have enable AWS Local zones , then it try to pick thoses AZ by default and failed to provision.
This should be applied to all pattern by default

Reply:
its a data source and we have encountered this issue with LZs before so ok to add